### PR TITLE
Collected small changes

### PR DESCRIPTION
**Pull request by akohlmey**
**Summary**

This pull request combines multiple small changes that do not warrant a PR on their own.

**Author(s)**

Axel Kohlmeyer, Temple U

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

No known issues.

**Implementation Notes**

The following individual changes are included:
- add a "magic" file with patterns for use with the file(1) command to the tools folder. They can be added to either /etc/magic or ~/.magic and then file can detect and print information about restart, binary and text mode dumps, data files and log files.
- drop constexpr from fmtlib buffer definition without initializer
- propagate KOKKOS settings to compiling main.cpp to suppress warnings about deprecated cuda architectures from nvcc_wrapper
- enable workaround for problems with new GPU package neighborlist code with CUDA (works around some of the issues reported in #2656)
- silence compiler warnings
- update false positives for documentation spellcheck and reduce warnings from doxygen processing
- add a CMake preset for using Intel's new LLVM based compilers bundled with their oneAPI toolkit
- detect new Intel Compilers with their new compiler specific macros instead of misdetecting them as Clang

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The added/updated documentation is integrated and tested with the documentation build system
- [x] The feature has been verified to work with the conventional build system
- [x] The feature has been verified to work with the CMake based build system
- [x] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

### CODE REVIEW

**Comment by akohlmey** [comment]
@sjplimp I am stunned to realize that nobody wrote some entries for the "magic" file for LAMMPS restart/dump files before (*and* made those available).

**Comment by rbberger** [comment]
@akohlmey should the oneAPI preset use the "new" Intel Fortran compiler `ifx`?

**Comment by akohlmey** [comment]
> @akohlmey should the oneAPI preset use the "new" Intel Fortran compiler `ifx`?

thanks for the pointer. Should be corrected now. I **had** looked for a matching fortran compiler, but somehow missed it. Not that it matters much these days for LAMMPS.

File: doc/src/Build_basics.rst [hunk 1]
```diff
@@ -234,6 +234,8 @@ LAMMPS.
          cmake ../cmake -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_Fortran_COMPILER=gfortran
          # Building with Intel Compilers:
          cmake ../cmake -DCMAKE_C_COMPILER=icc -DCMAKE_CXX_COMPILER=icpc -DCMAKE_Fortran_COMPILER=ifort
+         # Building with Intel oneAPI Compilers:
+         cmake ../cmake -DCMAKE_C_COMPILER=icx -DCMAKE_CXX_COMPILER=icpx -DCMAKE_Fortran_COMPILER=ifort
```
Comment (rbberger) [inline doc/src/Build_basics.rst:238]: missed one
```suggestion
         cmake ../cmake -DCMAKE_C_COMPILER=icx -DCMAKE_CXX_COMPILER=icpx -DCMAKE_Fortran_COMPILER=ifx
```

**Review by sjplimp — APPROVED** [review]

**Pull request merged**